mackerel-plugin-gcp-compute-engine/lib/gcp-compute-engine.go

```diff
@@ -0,0 +1,202 @@
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+	"strings"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+
+	"google.golang.org/api/monitoring/v3"
+
+	mp "github.com/mackerelio/go-mackerel-plugin"
+)
+
+const zuluFormat string = "2006-01-02T15:4:05Z"
+const duration string = "3m0s"
+const computeDomain string = "compute.googleapis.com"
+
+// ComputeEnginePlugin is mackerel plugin for Google Compute Engine
+type ComputeEnginePlugin struct {
+	Project           string
+	InstanceName      string
+	MonitoringService *monitoring.Service
+	Option            *Option
+}
+
+// Option is optional argument to an API call
+type Option struct {
+	Key string
+}
+
+// Get returns key and value
+func (c Option) Get() (string, string) {
+	return "key", c.Key
+}
+
+var graphdef = map[string]mp.Graphs{
+	"Firewall.DroppedBytesCount": mp.Graphs{
+		Label: "FireWall Dropped Bytes Count",
+		Unit:  "bytes",
+		Metrics: []mp.Metrics{
+			{Name: "dropped_bytes_count", Label: "Dropped Bytes Count"},
+		},
+	},
+	"Firewall.DroppedPacketsCount": mp.Graphs{
+		Label: "FireWall Dropped Packets Count",
+		Unit:  "float",
+		Metrics: []mp.Metrics{
+			{Name: "dropped_packets_count", Label: "Dropped Packets Count"},
+		},
+	},
+	"Cpu.Utilization": mp.Graphs{
+		Label: "CPU Utilization",
+		Unit:  "percentage",
+		Metrics: []mp.Metrics{
+			{Name: "utilization", Label: "Utilization"},
+		},
+	},
+	"Disk.BytesCount": mp.Graphs{
+		Label: "Disk Read Bytes Count",
+		Unit:  "bytes",
+		Metrics: []mp.Metrics{
+			{Name: "read_bytes_count", Label: "Read Bytes Count"},
+			{Name: "write_bytes_count", Label: "Write Bytes Count"},
+		},
+	},
+	"Disk.OpsCount": mp.Graphs{
+		Label: "Disk Read Ops Count",
+		Unit:  "float",
+		Metrics: []mp.Metrics{
+			{Name: "read_ops_count", Label: "Read Ops Count"},
+			{Name: "write_ops_count", Label: "Write Ops Count"},
+		},
+	},
+	"Network.BytesCount": mp.Graphs{
+		Label: "Network Received Bytes Count",
+		Unit:  "bytes",
+		Metrics: []mp.Metrics{
+			{Name: "received_bytes_count", Label: "Received Bytes Count"},
+			{Name: "sent_bytes_count", Label: "Sent Bytes Count"},
+		},
+	},
+	"Network.PacketsCount": mp.Graphs{
+		Label: "Network Received Packets Count",
+		Unit:  "float",
+		Metrics: []mp.Metrics{
+			{Name: "received_packets_count", Label: "Received Packets Count"},
+			{Name: "sent_packets_count", Label: "Sent Packets Count"},
+		},
+	},
+}
+
+// GraphDefinition is return graphdef
+func (p ComputeEnginePlugin) GraphDefinition() map[string]mp.Graphs {
+	return graphdef
+}
+
+func getLatestValue(listCall *monitoring.ProjectsTimeSeriesListCall, filter string, startTime string, endTime string, opts *Option) (float64, error) {
+	res, err := listCall.Filter(filter).IntervalEndTime(endTime).IntervalStartTime(startTime).Do(*opts)
+
+	if err != nil || res == nil {
+		return 0, err
+	}
+
+	valuePtr := res.TimeSeries[0].Points[0].Value
+
+	var value float64
+	if valuePtr.Int64Value != nil {
+		value = float64(*valuePtr.Int64Value)
```

Comment:
[FYI]
If you use mackerel-plugin-helper, Int64 values are accepted therefore you don't have to cast Int64value to float64: https://github.com/mackerelio/go-mackerel-plugin-helper/blob/master/mackerel-plugin.go#L38